### PR TITLE
Simple Firebase Push Notification

### PR DESCRIPTION
**Pull request by s3ppo**

### CODE REVIEW

**Comment by s3ppo** [comment]
done

**Review by Meldiron — CHANGES_REQUESTED** [review]
Before I review, please check "files changed" tab. There are 2 folders with source code - `push-nitification` and `PushNotification`. Which one is correct? Can we please remove one of them?

**Comment by s3ppo** [comment]
sry my bad, i forgot to remove the old folder :(
please try again

File: nodejs/push-notification/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+```bash
+$ cd ..
+$ tar -zcvf code.tar.gz push-notification
+```
+
+* Navigate to the Overview Tab of your Cloud Function > Deploy Tag
+* Input the command that will run your function (in this case "node index.js") as your entrypoint command
+* Upload your tarfile 
+* Click 'Activate'
```
Comment (christyjacob4) [inline nodejs/push-notification/README.md:37]: Please add a triggers section like the other examples. You can use this as a reference https://github.com/appwrite/demos-for-functions/tree/master/nodejs/object-detection#-trigger

File: nodejs/push-notification/index.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+//Perform all your imports
+const sdk = require('node-appwrite');
+const adm = require("firebase-admin");
+
+//place your firebase exported serviceAccountKey.json in same folder where your index.js is
+var serviceAccount = require("/usr/local/src/serviceAccountKey.json");
```
Comment (christyjacob4) [inline nodejs/push-notification/index.js:6]: Any reason you used the absolute path ? 
Isn't it better to use relative path like `./serviceAccountKey.json` ? 🤔

File: nodejs/push-notification/index.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+adm.initializeApp({
+    credential: adm.credential.cert(serviceAccount),
+    databaseURL: "<place_your_firebase_url_here>"
+});
+
+// Initialise the client SDK
+let client = new sdk.Client();
+client
+    .setEndpoint(process.env.APPWRITE_ENDPOINT)
```
Comment (christyjacob4) [inline nodejs/push-notification/index.js:16]: The endpoint needs to be included as an environment variable and needs to be mentioned in the environment variables section of  your README.

File: nodejs/push-notification/index.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    .setProject(process.env.APPWRITE_PROJECT_ID)
+    .setKey(process.env.APPWRITE_API_KEY);
```
Comment (christyjacob4) [inline nodejs/push-notification/index.js:18]: The Key needs to be included as an environment variable and needs to be mentioned in the environment variables section of  your README.

File: nodejs/push-notification/index.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+//Perform all your imports
+const sdk = require('node-appwrite');
+const adm = require("firebase-admin");
+
+//place your firebase exported serviceAccountKey.json in same folder where your index.js is
+var serviceAccount = require("/usr/local/src/serviceAccountKey.json");
+
+adm.initializeApp({
+    credential: adm.credential.cert(serviceAccount),
+    databaseURL: "<place_your_firebase_url_here>"
```
Comment (christyjacob4) [inline nodejs/push-notification/index.js:10]: The Firebase URL needs to be included as an environment variable and needs to be mentioned in the environment variables section of  your README.

File: nodejs/push-notification/index.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    .setEndpoint(process.env.APPWRITE_ENDPOINT)
+    .setProject(process.env.APPWRITE_PROJECT_ID)
+    .setKey(process.env.APPWRITE_API_KEY);
+
+const payload = JSON.parse(process.env.APPWRITE_FUNCTION_DATA);
+const title = payload['title'];
+const body = payload['body'];
+const sender = payload['sender'];
+const message = payload['message'];
+const token = "<place_user_token_here>";
```
Comment (christyjacob4) [inline nodejs/push-notification/index.js:25]: Which token is this ? This needs to be passed as an env variable or as part of the `FUNCTION_DATA` whichever is most appropriate

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Please address the comments

**Comment by christyjacob4** [comment]
@s3ppo  closing this PR for now. Please feel free to reopen it/ create a new one when you are ready to work on it once again

**Pull request closed without merging**